core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -0,0 +1,127 @@
+      return false;
+    }
+    return true;
+  }
+
+  private boolean validatePrincipalLogin() {
+    // Check whether can login with specified principal and keytab
+    String principal = "";
+    String keytab = "";
+    String primary;
+    String instance;
+    String realm;
+    if (mProcess.equals("master")) {
+      principal = Configuration.get(PropertyKey.MASTER_PRINCIPAL);
+      keytab = Configuration.get(PropertyKey.MASTER_KEYTAB_KEY_FILE);
+    } else if (mProcess.equals("worker")) {
+      principal = Configuration.get(PropertyKey.WORKER_PRINCIPAL);
+      keytab = Configuration.get(PropertyKey.WORKER_KEYTAB_FILE);
+    }
+    Matcher matchPrincipal = PRINCIPAL_PATTERN.matcher(principal);
+    if (!matchPrincipal.matches()) {
+      System.err.format("Principal %s is not in the right format.%n", principal);
+      return false;
+    }
+    primary = matchPrincipal.group("primary");
+    instance = matchPrincipal.group("instance");
+    realm = matchPrincipal.group("realm");
+
+    // Login with principal and keytab
+    int exitVal =
+        Utils.getResultFromProcess(new String[] {"kinit", "-kt", keytab, principal}).getExitValue();
+    if (exitVal != 0) {
+      System.err.format("Kerberos login failed for %s with keytab %s with exit value %d.%n",
+          principal, keytab, exitVal);
+      System.err.format("Primary is %s, instance is %s and realm is %s.%n",
+          primary, instance, realm);
+      return false;
+    }
+    return true;
+  }
+
+  private boolean validateHdfsSettingParity(Map<String, String> optionsMap) {
+    String serverHadoopConfDirPath;
+    if (optionsMap.containsKey(ValidateEnv.HADOOP_CONF_DIR_OPTION.getOpt())) {
+      serverHadoopConfDirPath = optionsMap.get(ValidateEnv.HADOOP_CONF_DIR_OPTION.getOpt());
+    } else {
+      serverHadoopConfDirPath = System.getenv(HADOOP_CONF_DIR_ENV_VAR);
+    }
+    if (serverHadoopConfDirPath == null) {
+      System.out.println("Path to server-side hadoop configuration unspecified,"
+          + " aborting validation for HDFS properties.");
+      return false;
+    }
+    String serverCoreSiteFilePath = serverHadoopConfDirPath + "/core-site.xml";
+    String serverHdfsSiteFilePath = serverHadoopConfDirPath + "/hdfs-site.xml";
+
+    // If Configuration does not contain the key, then a {@link RuntimeException} will be thrown
+    // before calling the {@link String#split} method.
+    String[] clientHaoopConfFilePaths = Configuration.get(PropertyKey.UNDERFS_HDFS_CONFIGURATION).split(":");
+    String clientCoreSiteFilePath = clientHaoopConfFilePaths[0];
+    String clientHdfsSiteFilePath = clientHaoopConfFilePaths[1];
+    if (clientCoreSiteFilePath == null || clientCoreSiteFilePath.isEmpty()
+        || clientHdfsSiteFilePath == null || clientHdfsSiteFilePath.isEmpty()) {
+      System.out.println("Cannot locate the client-side hadoop configurations,"
+          + " aborting validation for HDFS properties.");
+      return false;
+    }
+    return true;
```

Comment:
Can we also verify that the content of the hdfs server configuration files and the content of the alluxio hdfs configuration files are the same?

Reply:
Sure, will do.